### PR TITLE
Base64.Encode performance improvements

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
... at least when destination is aligned.

### CODE REVIEW

File: src/System.Binary.Base64/System/Binary/Base64.cs [hunk 1]
```diff
@@ -43,6 +45,30 @@ public static void Encode(byte b0, byte b1, byte b2, out byte r0, out byte r1, o
             r3 = s_encodingMap[i3];
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int Encode(ReadOnlySpan<byte> threeBytes)
+        {
+            byte b0 = threeBytes[0];
```
Comment (bartonjs) [inline src/System.Binary.Base64/System/Binary/Base64.cs]: Debug.Assert(threeBytes.Length >= 3)?

File: src/System.Binary.Base64/System/Binary/Base64.cs [hunk 2]
```diff
@@ -43,6 +45,30 @@ public static void Encode(byte b0, byte b1, byte b2, out byte r0, out byte r1, o
...
+            byte b1 = threeBytes[1];
+            byte b2 = threeBytes[2];
+
+            int i3 = b2 & 63;
```
Comment (bartonjs) [inline src/System.Binary.Base64/System/Binary/Base64.cs]: Personally, I like bitmasks in hex

Comment (benaadams) [inline src/System.Binary.Base64/System/Binary/Base64.cs]: `int i3 = b2 & 0b_0011_1111` :trollface:

File: src/System.Binary.Base64/System/Binary/Base64.cs (hunk 1 above)
Comment (KrzysztofCwalina) [inline src/System.Binary.Base64/System/Binary/Base64.cs]: Will do.

File: src/System.Binary.Base64/System/Binary/Base64.cs (hunk 2 above)
Comment (KrzysztofCwalina) [inline src/System.Binary.Base64/System/Binary/Base64.cs]: Will do. Yeah, the binary syntax would be great, but not yet available.

**Pull request merged**